Userland/Libraries/LibGfx/MedianCut.cpp

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright (c) 2024, Lucas Chollet <lucas.chollet@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/QuickSort.h>
+#include <AK/Statistics.h>
+#include <LibGfx/MedianCut.h>
+
+namespace Gfx {
+
+namespace {
+
+using Bucket = Vector<ARGB32>;
+using Buckets = Vector<Bucket>;
+
+void sort_along_color(Bucket& bucket, u8 color_index)
+{
+    auto less_than = [=](ARGB32 first, ARGB32 second) {
+        auto const first_color = Color::from_argb(first);
+        auto const second_color = Color::from_argb(second);
+        switch (color_index) {
+        case 0:
+            return first_color.red() < second_color.red();
+        case 1:
+            return first_color.green() < second_color.green();
+        case 2:
+            return first_color.blue() < second_color.blue();
+        default:
+            VERIFY_NOT_REACHED();
+        }
+    };
+
+    AK::quick_sort(bucket, less_than);
+}
+
+template<typename T>
+struct MaxAndIndex {
+    T maximum;
+    u32 index;
+};
+
+template<typename T, class MoreThan>
+MaxAndIndex<T> max_and_index(Span<T> values, MoreThan more_than)
```

Comment:
Hm, this is called `argmax` in numpy and `std::max_element` in the c++ stl (…ish. they just give you the index, but it's easy to get the value when you have the index.)

I couldn't quickly find something like this in AK, but this arguably belongs in AK. (Doesn't have to be in this PR.)

Reply:
I was also surprised that we don't already have it. Let's keep it here for the moment, I will move to AK later (this will even give us time to bikeshed the name on Discord in the meantime.)